### PR TITLE
chore: adds release process.

### PR DESCRIPTION
**Pull request by jcchavezs**
This PR adds the release script for the Go agent.

Ping @mohit-a21 @davexroth @pavolloffay

### CODE REVIEW

**Comment by mohit-a21** [comment]
@jcchavezs Can you outline the release workflow? In what case, one will have to run the script?

File: README.md [hunk 1]
```diff
@@ -169,3 +169,7 @@ for unit tests only
 ```bash
 make test-unit
 ```
+
+### Releasing
+
+Run `./release.sh <version_number>`.
```
Comment (pavolloffay) [inline README.md:175]: what is the release number? An example would be helpful

File: README.md [hunk 2]
```diff
@@ -169,3 +169,7 @@ for unit tests only
 ```bash
 make test-unit
 ```
+
+### Releasing
```
Comment (pavolloffay) [inline README.md:173]: Could you please explain by words what the release script does?

File: release.sh [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+git tag -a "$VERSION" -m "Version $VERSION"
+
+NEW_VERSION="$MAJOR.$MINOR.$(($PATCH+1))-dev"
+
+write_version_file $NEW_VERSION $VERSION_FILE
+git add $VERSION_FILE
+
+git commit -m "chore(version): prepares for next version $NEW_VERSION"
+
+git push origin $MAIN_BRANCH || { rollback_changes $VERSION ; echo "Failed to push to $MAIN_BRANCH" ; exit 1 }
```
Comment (pavolloffay) [inline release.sh:83]: Many people put origin as their for (including myself). Hence the should be configurable or use the full repo path (forked repos usually configure release separately anyways)

File: release.sh [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+git push --tags
```
Comment (pavolloffay) [inline release.sh:85]: missing remote?

File: README.md (hunk 2 above)
Comment (jcchavezs) [reply README.md:173]: Sure.

File: release.sh (hunk 3 above)
Comment (jcchavezs) [reply release.sh:83]: As of now, only maintainers can do a release (and none of us need a fork) and in that case I would keep this script simple: "convention over configuration"

File: release.sh (hunk 4 above)
Comment (jcchavezs) [reply release.sh:85]: Not sure what you mean, do you think this isn't enough?

File: release.sh (hunk 3 above)
Comment (pavolloffay) [reply release.sh:83]: I use always forks!!!!

File: release.sh (hunk 4 above)
Comment (pavolloffay) [reply release.sh:85]: To which remote does the push pushes tags? I use always forks and use origin for my fork.

File: release.sh (hunk 3 above)
Comment (pavolloffay) [reply release.sh:83]: Keep the upstream clean ;)

File: release.sh (hunk 4 above)
Comment (jcchavezs) [reply release.sh:85]: As I said, I rather keep this simple and asume the releaser uses `origin` (in practice it is going to be me as of now). Later we can solve the fork thing.

**Pull request merged**